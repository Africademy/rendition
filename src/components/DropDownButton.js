import styled, { withTheme } from 'styled-components'
import { px } from '../utils'
import IconCaretDown from 'react-icons/lib/fa/caret-down'
import IconCaretUp from 'react-icons/lib/fa/caret-up'
import { h, Component } from 'preact'
import Button from './Button'
import Fixed from './Fixed'
import { Box, Flex } from './Grid'
import { compose } from 'recompose'
import { space, color, fontSize, width } from 'styled-system'

const ToggleBase = styled(Button)`
  min-width: 0;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  margin-left: -1px;
  border-left: 0;
  margin: 0;
  border-width: ${props => props.outline && '1px'};
`

const ButtonBase = styled(Button)`
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right: 0;
  margin: 0;
  border-width: ${props => props.outline && '1px'};
`

const MenuBase = styled.div`
  background: white;
  position: absolute;
  min-width: ${props => props.minWidth};
  box-shadow: ${props => '1px 1px 5px' + props.theme.colors.gray.light};
  border-radius: ${props => px(props.theme.radius)};
  border: ${props => '1px solid ' + props.theme.colors.gray.main};
  overflow: hidden;
  z-index: 1;
  left: ${props => (props.alignRight ? 'auto' : 0)};
  right: ${props => (!props.alignRight ? 'auto' : 0)};
`

const Wrapper = styled.div`
  ${space} ${width} ${fontSize} ${color} display: inline-block;
  vertical-align: top;
  position: relative;
`

const Item = styled.div`
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 16px;
  padding-right: 16px;
  border-top: ${props =>
    props.border && '1px solid ' + props.theme.colors.gray.main};
  white-space: nowrap;

  &:hover {
    background: ${props => props.theme.colors.gray.light};
  }
`

const IconWrapper = styled.span`width: 28px;`

const JoinedButton = styled(Button)`margin: 0;`

const Toggle = ({ open, handler, label, joined, ...props }) => {
  if (joined) {
    if (label) {
      return (
        <JoinedButton {...props} pl={16} pr={0} onClick={handler}>
          <Flex justify='space-between'>
            <Box mt='1px'>{label}</Box>
            <IconWrapper>
              {open ? <IconCaretUp /> : <IconCaretDown />}
            </IconWrapper>
          </Flex>
        </JoinedButton>
      )
    }
    return (
      <JoinedButton {...props} square onClick={handler}>
        {open ? <IconCaretUp /> : <IconCaretDown />}
      </JoinedButton>
    )
  }
  return (
    <ToggleBase {...props} onClick={handler}>
      {open ? <IconCaretUp /> : <IconCaretDown />}
    </ToggleBase>
  )
}

class DropDownButton extends Component {
  constructor (props) {
    super(props)
    this.state = {
      open: false,
      minWidth: 0
    }
  }

  toggle (e) {
    this.setState({
      open: !this.state.open,
      minWidth: this.base && this.base.offsetWidth
    })
  }

  render ({ alignRight, children, label, border, joined, ...props }) {
    return (
      <Wrapper {...props}>
        {joined ? (
          <Toggle
            {...props}
            joined={joined}
            label={label}
            handler={e => this.toggle(e)}
            open={this.state.open}
          />
        ) : (
          <span>
            <ButtonBase {...props}>{label}</ButtonBase>
            <Toggle
              {...props}
              handler={e => this.toggle(e)}
              open={this.state.open}
            />
          </span>
        )}
        {this.state.open && <Fixed onClick={e => this.toggle(e)} />}
        {this.state.open && (
          <MenuBase
            alignRight={alignRight}
            onClick={e => this.toggle(e)}
            minWidth={`${this.state.minWidth}px`}
          >
            {children.map((child, i) => {
              return (
                <Item border={border && i} key={i}>
                  {child}
                </Item>
              )
            })}
          </MenuBase>
        )}
      </Wrapper>
    )
  }
}

export default compose(withTheme)(DropDownButton)
