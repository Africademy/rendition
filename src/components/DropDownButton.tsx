import * as React from 'react';
import IconCaretDown = require('react-icons/lib/fa/caret-down');
import IconCaretUp = require('react-icons/lib/fa/caret-up');
import styled from 'styled-components';
import asRendition from '../asRendition';
import { RenditionSystemProps } from '../common-types';
import { px } from '../utils';
import Button, { ButtonProps } from './Button';
import Divider from './Divider';
import Fixed from './Fixed';
import { Box, Flex } from './Grid';

const ToggleBase = styled(Button)`
	min-width: 0;
	border-top-left-radius: 0;
	border-bottom-left-radius: 0;
	margin-left: -1px;
	border-left: 0;
	margin: 0;
	vertical-align: top;
`;

const ButtonBase = styled(Button)`
	border-top-right-radius: 0;
	border-bottom-right-radius: 0;
	border-right: 0;
	margin: 0;
`;

const MenuBase = styled.div<{
	minWidth: string;
	alignRight?: boolean;
	maxHeight: string | number;
}>`
	background: white;
	position: absolute;
	min-width: ${props => props.minWidth};
	box-shadow: ${props => '1px 1px 5px' + props.theme.colors.gray.light};
	border-radius: ${props => px(props.theme.radius)};
	border: ${props => '1px solid ' + props.theme.colors.gray.main};
	z-index: 1;
	margin-top: 2px;
	left: ${props => (props.alignRight ? 'auto' : 0)};
	right: ${props => (!props.alignRight ? 'auto' : 0)};
	white-space: nowrap;
	max-height: ${props => props.maxHeight || 'auto'};
	overflow-y: auto;
`;

const Wrapper = styled(Box)`
	display: inline-block;
	border-radius: ${props => px(props.theme.radius)};
	vertical-align: top;
	position: relative;
`;

const Item = styled.div<{ border: boolean }>`
	padding-top: 5px;
	padding-bottom: 5px;
	padding-left: 16px;
	padding-right: 16px;
	border-top: ${props =>
		props.border ? `1px solid ${props.theme.colors.gray.main}` : '0'};
	border-radius: ${props => px(props.theme.radius)};

	&:hover:enabled {
		background: ${props => props.theme.colors.gray.light};
	}
`;

const IconWrapper = styled.span`
	width: 28px;
`;

const JoinedButton = styled(Button)`
	margin: 0;
`;

const Toggle = ({
	open,
	handler,
	label,
	joined,
	...props
}: InternalDropDownButtonProps & {
	open: boolean;
	handler: React.MouseEventHandler<HTMLElement>;
}) => {
	if (joined) {
		if (label) {
			return (
				<JoinedButton {...props} pl={16} pr={0} onClick={handler}>
					<Flex justifyContent="space-between" alignItems="center">
						<Box mt="1px">{label}</Box>
						<IconWrapper>
							{open ? <IconCaretUp /> : <IconCaretDown />}
						</IconWrapper>
					</Flex>
				</JoinedButton>
			);
		}
		return (
			<JoinedButton {...props} square onClick={handler}>
				{open ? <IconCaretUp /> : <IconCaretDown />}
			</JoinedButton>
		);
	}
	return (
		<ToggleBase {...props} onClick={handler}>
			{open ? <IconCaretUp /> : <IconCaretDown />}
		</ToggleBase>
	);
};

interface DropDownButtonState {
	open: boolean;
	minWidth: number;
}

class DropDownButton extends React.Component<
	InternalDropDownButtonProps,
	DropDownButtonState
> {
	dropdownNode: HTMLDivElement | null;

	constructor(props: InternalDropDownButtonProps) {
		super(props);
		this.state = {
			open: false,
			minWidth: 0,
		};
	}

	componentWillUnmount() {
		if (this.state.open) {
			this.removeListener();
		}
	}

	handleOutsideClick = (e: MouseEvent) => {
		if (this.dropdownNode && !this.dropdownNode.contains(e.target as Node)) {
			if (this.state.open) {
				this.removeListener();
				this.setState({
					open: false,
				});
			}
		}
	};

	toggle = () => {
		const isNextOpen = !this.state.open;
		if (isNextOpen) {
			window.document.addEventListener('click', this.handleOutsideClick);
		} else {
			this.removeListener();
		}

		this.setState({
			open: isNextOpen,
			minWidth: (this as any).base && (this as any).base.offsetWidth,
		});
	};

	removeListener = () => {
		window.document.removeEventListener('click', this.handleOutsideClick);
	};

	render() {
		const {
			alignRight,
			children,
			label,
			border,
			joined,
			noListFormat,
			outline,
			className,
			...props
		} = this.props;

		return (
			<Wrapper
				className={className}
				{...props}
				ref={node => (this.dropdownNode = node)}
			>
				{joined ? (
					<Toggle
						{...props}
						outline={outline}
						joined={joined}
						label={label}
						handler={this.toggle}
						open={this.state.open}
					/>
				) : (
					<span>
						<ButtonBase {...props} outline={outline}>
							{label}
						</ButtonBase>
						<Toggle
							{...props}
							outline={outline}
							handler={this.toggle}
							open={this.state.open}
						/>
					</span>
				)}
				{this.state.open && <Fixed onClick={this.toggle} />}
				{this.state.open && (
					<MenuBase
						alignRight={alignRight}
						onClick={this.toggle}
						minWidth={`${this.state.minWidth}px`}
						maxHeight={px(this.props.maxHeight)}
					>
						{React.Children.map(children, (child, i) => {
							if (noListFormat) {
								return child;
							}
							if (!child) {
								return;
							}
							if ((child as any).type === Divider) {
								return child;
							}
							return (
								<Item border={Boolean(border && i)} key={i}>
									{child}
								</Item>
							);
						})}
					</MenuBase>
				)}
			</Wrapper>
		);
	}
}

export interface InternalDropDownButtonProps extends ButtonProps {
	label?: JSX.Element;
	border?: boolean;
	joined?: boolean;
	alignRight?: boolean;
	noListFormat?: boolean;
	maxHeight?: string | number;
}

export type DropDownButtonProps = InternalDropDownButtonProps &
	RenditionSystemProps;

export default asRendition<
	React.ForwardRefExoticComponent<
		DropDownButtonProps & React.RefAttributes<DropDownButton>
	>
>(DropDownButton, [], ['width', 'color', 'bg', 'backgroundColor']);
