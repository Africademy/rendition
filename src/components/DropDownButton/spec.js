/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import { DropDownButton, Provider } from '../../../dist'

test('DropDownButton renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <DropDownButton primary label={<div>DropDown</div>}>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
      </DropDownButton>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Opened DropDownButton renders correctly', () => {
  let componentInstance
  const component = renderer.create(
    <Provider>
      <DropDownButton
        primary
        ref={(ref) => {
          componentInstance = ref
        }}
        label={<div>DropDown</div>}
      >
        <div>Item</div>
        {[...Array(3).keys()].map((index) => (
          <div key={index}>Item</div>
        ))}
      </DropDownButton>
    </Provider>
  )

  componentInstance.setState({
    open: true
  })

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
