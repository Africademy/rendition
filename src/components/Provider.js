import * as React from 'react'
import styled, { ThemeProvider } from 'styled-components'
import defaultTheme from '../theme'

const Base = styled.div`
  font-family: ${props => props.theme.font};
`

const Provider = ({ theme, ...props }) => {
  return (
    <ThemeProvider theme={theme || defaultTheme}>
      <Base {...props} />
    </ThemeProvider>
  )
}

export default Provider
