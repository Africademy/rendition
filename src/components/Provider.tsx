import * as React from 'react';
import styled, { StyledFunction, ThemeProvider } from 'styled-components';
import { DefaultProps, Theme } from '../common-types';
import defaultTheme from '../theme';

export interface ThemedBaseProps extends DefaultProps {
	theme?: Theme;
}

const Base = (styled.div as StyledFunction<DefaultProps>)`
  font-family: ${props => props.theme.font};
`;

const Provider = ({ theme, ...props }: ThemedBaseProps) => {
	return (
		<ThemeProvider theme={theme || defaultTheme}>
			<Base {...props} />
		</ThemeProvider>
	);
};

export default Provider;
