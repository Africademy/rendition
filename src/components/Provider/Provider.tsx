import {
	Grommet
} from 'grommet'
import * as React from 'react';

const Provider = ({ theme, ...props }: any) => {
	return (
		<Grommet theme={theme} {...props} />
	);
};

export default Provider;
