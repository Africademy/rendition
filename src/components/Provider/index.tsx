import { Grommet } from 'grommet';
import merge from 'lodash/merge';
import * as React from 'react';
import styled from 'styled-components';
import { DefaultProps, Theme } from '../../common-types';
import defaultTheme from '../../theme';
import { px } from '../../utils';
import { BreakpointProvider } from './BreakpointProvider';

const Base = styled(Grommet)`
	font-family: ${props => props.theme.font};
	font-size: ${props => px(props.theme.fontSizes[1])};

	h1,
	h2,
	h3,
	h4,
	h5,
	h6,
	button {
		font-family: ${props => props.theme.titleFont};
	}
`;

const Provider = ({ theme, ...props }: ThemedProvider) => {
	const providerTheme = merge(defaultTheme, theme);
	return (
		<BreakpointProvider breakpoints={providerTheme.breakpoints}>
			<Base theme={providerTheme} {...props} />
		</BreakpointProvider>
	);
};

export interface ThemedProvider extends DefaultProps {
	theme?: Partial<Theme>;
}

export default Provider;
