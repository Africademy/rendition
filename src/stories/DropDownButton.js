import { h } from 'preact'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'
import DropDownButton from '../components/DropDownButton'
import { Flex } from '../components/Grid'
import Divider from '../components/Divider'

const Container = styled.div`margin: 30px;`

storiesOf('DropDownButton', module)
  .addWithInfo('Standard', () => {
    return (
      <Container>
        <DropDownButton mx={2} primary label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} secondary label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} tertiary label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
      </Container>
    )
  })
  .addWithInfo('Borderless', () => {
    return (
      <Container>
        <DropDownButton
          mx={2}
          border={false}
          primary
          label={<div>DropDown</div>}
        >
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton
          mx={2}
          border={false}
          secondary
          label={<div>DropDown</div>}
        >
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton
          mx={2}
          border={false}
          tertiary
          label={<div>DropDown</div>}
        >
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} border={false} label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
      </Container>
    )
  })
  .addWithInfo('Joined', () => {
    return (
      <Container>
        <DropDownButton mx={2} joined primary label='DropDown'>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} joined secondary label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} joined tertiary label={<div>DropDown</div>}>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
        <DropDownButton mx={2} joined>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
          <div>Item</div>
        </DropDownButton>
      </Container>
    )
  })
  .addWithInfo('Alignment', () => {
    return (
      <Container>
        <Flex justify='space-between'>
          <DropDownButton mx={2} primary label={<div>DropDown</div>}>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          </DropDownButton>
          <DropDownButton
            mx={2}
            alignRight
            secondary
            label={<div>DropDown</div>}
          >
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
            <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          </DropDownButton>
        </Flex>
      </Container>
    )
  })
  .addWithInfo('Divider', () => {
    return (
      <Container>
        <DropDownButton mx={2} primary label={<div>DropDown</div>}>
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          <Divider color='#c6c8c9' height={1} />
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div>
        </DropDownButton>
      </Container>
    )
  })
