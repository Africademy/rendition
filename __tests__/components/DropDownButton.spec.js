/* globals expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import DropDownButton from '../../src/components/DropDownButton'

test('DropDownButton renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <DropDownButton primary label={<div>DropDown</div>}>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
      </DropDownButton>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Opened DropDownButton renders correctly', () => {
  let componentInstance
  const component = renderer.create(
    <Provider>
      <DropDownButton primary ref={ref => { componentInstance = ref }} label={<div>DropDown</div>}>
        <div>Item</div>
        {[...Array(3).keys()].map((index) => (
          <div key={index}>Item</div>
        ))}
      </DropDownButton>
    </Provider>
  )

  componentInstance.setState({
    open: true
  })

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
