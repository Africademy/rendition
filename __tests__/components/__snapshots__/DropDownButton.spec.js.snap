// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownButton renders correctly 1`] = `
<div
  className="sc-cmTdod ePZXKJ StyledGrommet-sc-19lkkz7-0 bBubtL"
>
  <div
    className="sc-hqyNC hbagY sc-ksYbfQ hqnrsh sc-gzVnrw pznRG sc-bZQynM kMvugX"
  >
    <span>
      <button
        className="sc-kEYyzF sc-kgAjT JxtKU sc-cvbbAY ldJHua sc-kkGfuU kqxvkn StyledButton-sc-323bzc-0 jfOEyc"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <div>
          DropDown
        </div>
      </button>
      <button
        className="sc-kEYyzF sc-TOsTZ bJoXdg sc-cvbbAY ldJHua sc-kkGfuU kqxvkn StyledButton-sc-323bzc-0 dQGtHJ"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": "#444444",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m31.4 15.7q0 0.6-0.5 1l-10 10q-0.4 0.4-1 0.4t-1-0.4l-10-10q-0.4-0.4-0.4-1t0.4-1 1-0.4h20q0.6 0 1 0.4t0.5 1z"
            />
          </g>
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`Opened DropDownButton renders correctly 1`] = `
<div
  className="sc-cmTdod ePZXKJ StyledGrommet-sc-19lkkz7-0 bBubtL"
>
  <div
    className="sc-hqyNC hbagY sc-ksYbfQ hqnrsh sc-gzVnrw pznRG sc-bZQynM kMvugX"
  >
    <span>
      <button
        className="sc-kEYyzF sc-kgAjT JxtKU sc-cvbbAY ldJHua sc-kkGfuU kqxvkn StyledButton-sc-323bzc-0 jfOEyc"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <div>
          DropDown
        </div>
      </button>
      <button
        className="sc-kEYyzF sc-TOsTZ bJoXdg sc-cvbbAY ldJHua sc-kkGfuU kqxvkn StyledButton-sc-323bzc-0 dQGtHJ"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": "#444444",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m31.4 27.1q0 0.6-0.5 1t-1 0.5h-20q-0.6 0-1-0.5t-0.4-1 0.4-1l10-10q0.4-0.4 1-0.4t1 0.4l10 10q0.5 0.5 0.5 1z"
            />
          </g>
        </svg>
      </button>
    </span>
    <div
      className="sc-cHGsZl bsJgGW sc-kjoXOD hMWlni"
      onClick={[Function]}
    />
    <div
      className="sc-cJSrbW fZFKOk"
      onClick={[Function]}
    >
      <div
        className="sc-hmzhuo cRrTEJ"
      >
        <div>
          Item
        </div>
      </div>
      <div
        className="sc-hmzhuo cRrTEJ"
      >
        <div>
          Item
        </div>
      </div>
      <div
        className="sc-hmzhuo cRrTEJ"
      >
        <div>
          Item
        </div>
      </div>
      <div
        className="sc-hmzhuo cRrTEJ"
      >
        <div>
          Item
        </div>
      </div>
    </div>
  </div>
</div>
`;
